opendatahub-io/data-science-pipelines
# UPSTREAM: <carry>: Update ArtifactView enum to allow pipelines artifacts api to have optional content disposition

**Description of your changes:**
This PR resolves # [RHOAIENG-12014](https://issues.redhat.com/browse/RHOAIENG-12014)
- This PR improves the Pipelines Artifacts API by adding flexible URL options for artifact retrieval. 
- It introduces a new RENDER field to the enum, which, when selected, provides a signed URL without the content disposition header. 
- Updated DOWNLOAD view to download the artifact instead of showing preview and then download.
- This allows for the immediate download of artifacts in some cases, while also enabling preview or rendering in other scenarios.

**Testing Instructions:**

-  Deploy DSPA using this API Server image - ```quay.io/opendatahub/ds-pipelines-api-server:pr-118```
-  Create a pipeline run and use the below api to verify the changes in artifact view options:

1.      For DOWNLOAD view:
    ```
       url - /apis/v2beta1/artifacts/1?view=DOWNLOAD
       Method : GET
     ```
      ```
    Sample Response:
   {
    "artifact_id": "1",
    "storage_provider": "s3",
    "storage_path": "iris-training-pipeline/70e74c8c-fbe7-41a8-a20e-0dfb84db5516/create-dataset/iris_dataset",
    "uri": "s3://mlpipeline/iris-training-pipeline/70e74c8c-fbe7-41a8-a20e-0dfb84db5516/create-dataset/iris_dataset",
    "download_url": "https://minio-sample-dspa.apps.vmudadla-1.dev.datahub.redhat.com/mlpipeline/iris-training-pipeline/70e74c8c-fbe7-41a8-a20e-0dfb84db5516/create-dataset/iris_dataset?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=MgVISopjwQeEKLWa%2F20241125%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20241125T171009Z&X-Amz-Expires=60&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3D%22iris-training-pipeline%2F70e74c8c-fbe7-41a8-a20e-0dfb84db5516%2Fcreate-dataset%2Firis_dataset%22&X-Amz-Signature=88261cc9cb8fd40a5f29b67ac7e4bb2068ae636cbaa2840282d7e32fde86611f",
    "namespace": "dspa",
    "artifact_type": "system.Dataset",
    "artifact_size": "5098",
    "created_at": "2024-11-22T22:26:42.147Z",
    "last_updated_at": "2024-11-22T22:26:42.147Z"
    }
     ```
   Open the download_url in a browser, and the artifact should automatically start downloading.
2.      For RENDER view:
    ```
       url - /apis/v2beta1/artifacts/1?view=RENDER
       Method : GET
     ```
      ```
    Sample Response:
   {
    "artifact_id": "1",
    "storage_provider": "s3",
    "storage_path": "iris-training-pipeline/70e74c8c-fbe7-41a8-a20e-0dfb84db5516/create-dataset/iris_dataset",
    "uri": "s3://mlpipeline/iris-training-pipeline/70e74c8c-fbe7-41a8-a20e-0dfb84db5516/create-dataset/iris_dataset",
    "namespace": "dspa",
    "artifact_type": "system.Dataset",
    "artifact_size": "5098",
    "created_at": "2024-11-22T22:26:42.147Z",
    "last_updated_at": "2024-11-22T22:26:42.147Z",
    "render_url": "https://minio-sample-dspa.apps.vmudadla-1.dev.datahub.redhat.com/mlpipeline/iris-training-pipeline/70e74c8c-fbe7-41a8-a20e-0dfb84db5516/create-dataset/iris_dataset?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=MgVISopjwQeEKLWa%2F20241125%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20241125T195517Z&X-Amz-Expires=60&X-Amz-SignedHeaders=host&response-content-disposition=inline&X-Amz-Signature=e543aeab6b98b8e82f79b45d5a24ab154949430da8c21db6c8024e5c1eef7d18"
}
     ```
   Open the render_url in a browser to view a preview of the artifact.

**Checklist:**
